### PR TITLE
Add small file workload test for scale

### PR DESCRIPTION
**Pull request by wusui**
Look for storage space leaks in Ceph Filesystems.

Signed-off-by: wusui <wusui@redhat.com>

### CODE REVIEW

File: tests/e2e/scale/test_scale_small_file_workload.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    else:
+        es = None
+
+    def teardown():
+        if es is not None:
+            es.cleanup()
+            time.sleep(10)
+
+    request.addfinalizer(teardown)
+    return es
```
Comment (Avilir) [inline tests/e2e/scale/test_scale_small_file_workload.py:88]: you can use the general fixure from  tests.conftest.py (line 3531)

**Review by Avilir — COMMENTED** [review]
@wusui , nice work, just one thing, why we are not collecting the Smallfile performance results ?
we are not interesting on them ?

File: tests/e2e/scale/test_scale_small_file_workload.py (hunk 1 above)
Comment (wusui) [reply tests/e2e/scale/test_scale_small_file_workload.py]: Done.  I have pushed a new version.

File: ocs_ci/ocs/small_file_workload.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+from ocs_ci.utility import templating
+from ocs_ci.ocs.utils import get_pod_name_by_pattern
+from ocs_ci.ocs.version import get_environment_info
+from ocs_ci.ocs import constants
+
+log = logging.getLogger(__name__)
+
+
+def smallfile_workload(
+    ripsaw, es, scale_leaks, file_size, files, threads, samples, interface
```
Comment (ramkiperiy) [inline ocs_ci/ocs/small_file_workload.py:28]: Please add doc strings for params

File: tests/conftest.py [hunk 3]
```diff
@@ -3622,3 +3622,10 @@ def teardown():
     es = ElasticSearch()
 
     return es
+
+
+def ceph_fs_use():
+    """
+    Bytes used value saved by cephfs small file scaling test.
+    """
+    return 0
```
Comment (ramkiperiy) [inline tests/conftest.py:3631]: Is this function is called somewhere? didn't get the purpose of it.

**Comment by ramkiperiy** [comment]
Nice work @wusui !!

File: tests/e2e/scale/test_scale_small_file_workload.py (hunk 1 above)
Comment (wusui) [reply tests/e2e/scale/test_scale_small_file_workload.py]: The performance tests are still run in tests/e2e/performance/test_small_file_workload.py   After this PR is done, and after the  Elastic search exception that I reported on issue 3705 gets resolved, I will adapt the performance small file test to also use the ocs_ci/ocs/small_file_workload.py functions.

**Comment by wusui** [comment]
Changes suggested by Ram have been made.

File: ocs_ci/ocs/small_file_workload.py (hunk 2 above)
Comment (wusui) [reply ocs_ci/ocs/small_file_workload.py:28]: Done.

File: tests/conftest.py (hunk 3 above)
Comment (wusui) [reply tests/conftest.py]: I got rid of this.

**Comment by wusui** [comment]
The jenkins failure is happening because there is a leak in cephfilesystem-metadata.

POOLS:
    POOL                                           ID     STORED      OBJECTS     USED        %USED     MAX AVAIL 
    ocs-storagecluster-cephblockpool                1      11 GiB       3.15k      33 GiB      0.64       1.7 TiB 
    ocs-storagecluster-cephfilesystem-metadata      2     202 MiB          81     606 MiB      0.01       1.7 TiB 
    ocs-storagecluster-cephfilesystem-data0         3       158 B           1      12 KiB         0       1.7 TiB 

The USED value appears to be not going back down.

**Comment by wusui** [comment]
I have reported this leak in Bugzilla 1918964

**Comment by wusui** [comment]
I have modified the leak checks to not report an error if some of the ceph filesystem metadata pool is consumed when the test is initialized.  Later runs have demonstrated to me that further leaks do not happen.

This now passes.   I have also closed the associated bugzilla that I filed a few days ago.

Please re-review so this can be merged

Thanks.

File: tests/e2e/scale/test_scale_small_file_workload.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        for entry in now_data:
+            log.info(f"{entry} now uses {now_data[entry]} bytes")
+            log.info(f"{entry} originally used {orig_data[entry]} bytes")
+            #
+            # Skip metadata check if the space used is still under 1G.
+            # cephfilesytem-metadata pool consumes some storage when
+            # first run, so the first increase is larger than 10M.
+            # Skip metadata check if the space used is still under 1G.
+            #
+            if entry.endswith("metadata") and now_data[entry] < 1000000000:
```
Comment (keesturam) [inline tests/e2e/scale/test_scale_small_file_workload.py:65]: 10M is a very less number to define a memory leak.

File: tests/e2e/scale/test_scale_small_file_workload.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    """
+    pytest.ceph_fs_use = get_cephfs_data()
+
+    def teardown():
+        """
+        After all the ripsaw creation and deltion runs have finished,
+        test to see if there was a leak in storage.
+        """
+        log.info("In scale_leaks teardown")
+        time.sleep(300)
```
Comment (keesturam) [inline tests/e2e/scale/test_scale_small_file_workload.py:53]: why have you defined a hard sleep for 300 seconds?

File: tests/e2e/scale/test_scale_small_file_workload.py (hunk 4 above)
Comment (wusui) [reply tests/e2e/scale/test_scale_small_file_workload.py:65]: What number do you thing would be appropriate here?

File: tests/e2e/scale/test_scale_small_file_workload.py (hunk 5 above)
Comment (wusui) [reply tests/e2e/scale/test_scale_small_file_workload.py:53]: We don't need this.  I removed it and I will rerun the jenkins test.

File: tests/e2e/scale/test_scale_small_file_workload.py (hunk 4 above)
Comment (keesturam) [reply tests/e2e/scale/test_scale_small_file_workload.py:64]: As discussed offline, let's benchmark the average memory consumption and then check for any leaks after workloads are run.

**Comment by wusui** [comment]
Please Rereview.  All the suggested changes have been made.

File: tests/e2e/scale/test_scale_small_file_workload.py (hunk 4 above)
Comment (wusui) [reply tests/e2e/scale/test_scale_small_file_workload.py]: Fixed in latest change.

**Review by Avilir — APPROVED** [review]
Nice work @wusui

**Review by tiffanyn108 — APPROVED** [review]
LGTM.

**Review by vasukulkarni — APPROVED** [review]

**Pull request merged**